drivers/hmis/spec/requests/hmis/ce/ce_opportunity_spec.rb

```diff
@@ -194,16 +194,46 @@
           response, result = post_graphql(**variables) { query }
           expect(response.status).to eq(200), result.inspect
 
-          active_referral_data = result.dig('data', 'ceOpportunity', 'activeReferral')
+          referral_data = result.dig('data', 'ceOpportunity', 'referral')
 
-          expect(active_referral_data).to include(
+          expect(referral_data).to include(
             'id' => active_referral.id.to_s,
             'status' => 'in_progress',
           )
         end
       end
     end
 
+    describe 'when the opportunity has several referrals' do
+      # opportunities are single-use, so there should only be one in-progress or accepted referral, but there could be many failed referrals.
+      let!(:rejected1) { create(:hmis_ce_referral, opportunity: opportunity, status: 'rejected', created_at: 1.day.ago) }
+      let!(:rejected2) { create(:hmis_ce_referral, opportunity: opportunity, status: 'rejected', created_at: 1.day.ago) }
+      let!(:rejected3) { create(:hmis_ce_referral, opportunity: opportunity, status: 'rejected', created_at: 1.day.ago) }
+
+      ['initialized', 'in_progress', 'accepted'].each do |status|
+        it 'returns the correct referral' do
```

Comment:
suggest adding status into test name so test case name isnt repeated:
```suggestion
        it "returns the #{status} referral" do
```